### PR TITLE
 feat: Support focus to checkbox, radio, button-toggle and slide-toggle

### LINKED ISSUE #1131: Support focus method on checkbox and radio inputs
#### Bug, feature request, or proposal:

Support focus method on checkbox and radio inputs
#### What is the expected behavior?

Calling focus method on MdCheckbox and MdRadioButton should change focus to these instance. Also other focusable items in browser should lose the focus.
#### What is the current behavior?

focus method does not exists on MdCheckbox and MdRadioButton thus get error such as `focus is not a function`
#### What are the steps to reproduce?

http://plnkr.co/edit/9RJ5DOuPMK8b3zPeR95J?p=preview
#### What is the use-case or motivation for changing an existing behavior?

Similar to MdInput there should be way to focus on elements. this can be used for focusing to element with validation error
#### Which versions of Angular, Material, OS, browsers are affected?

Was able to reproduce on both firefox and chrome.

### PR DESCRIPTION
**Pull request by tinayuangao**
Closes #1131 

R: @jelbourn

### CODE REVIEW

File: src/lib/radio/radio.spec.ts [hunk 1]
```diff
@@ -188,6 +188,23 @@ describe('MdRadio', () => {
       expect(radioNativeElements[0].classList).not.toContain('md-radio-focused');
     });
 
+    it('should focus individual radio buttons', () => {
+      let nativeRadioInput = <HTMLElement> radioNativeElements[0].querySelector('input');
+      radioInstances[0].focus();
+
+      expect(nativeRadioInput.classList).not.toContain('md-radio-focused');
+
+      dispatchEvent('focus', nativeRadioInput);
```
Comment (jelbourn) [inline src/lib/radio/radio.spec.ts]: Why are you dispatching a focus event here instead of calling `focus`?

Comment (tinayuangao) [inline src/lib/radio/radio.spec.ts]: Fixed the test. PTAL. Thanks!

**Comment by jelbourn** [comment]
LGTM

**Comment by jelbourn** [comment]
@tinayuangao can you rebase this PR?

**Pull request merged**